civihr_employee_portal/views/templates/views-view-table--Documents--block.tpl.php

```diff
@@ -96,66 +96,79 @@
-            <tr <?php if ($row_classes[$row_count] || $class) {
-                print 'class="' . implode(' ', $row_classes[$row_count]) . ' ' . $class . '"';
-              } ?>>
-                <?php
-                foreach ($row as $field => $content) {
-                  $class = '';
-                  $attribute = '';
-                  if (!empty($field_classes[$field][$row_count])) {
-                    $class = sprintf('class = "%s"', $field_classes[$field][$row_count]);
-                  }
-                  if (!empty($field_attributes[$field][$row_count])) {
-                    $attribute = drupal_attributes($field_attributes[$field][$row_count]);
-                  }
-
-                  printf('<td %s %s>%s</td>', $class, $attribute, $content);
-                } ?>
-              <td data-ct-spinner data-ct-spinner-id="document-<?php print $rowID; ?>">
-                <?php if ($row['status_id'] === 'awaiting upload'): ?>
-                  <button
-                    ng-show='!document.loadingModalData'
-                    ng-click="document.modalDocument('<?php print $rowID ?>', 'staff', 'edit')"
-                    class="btn btn-sm btn-default">
-                    <i class="fa fa-upload"></i>
-                    Upload
-                  </button>
-                <?php else: ?>
-                  <button
-                    ng-show='!document.loadingModalData'
-                    ng-click="document.modalDocument('<?php print $rowID ?>', 'staff', 'view')"
-                    class="btn btn-sm btn-default">
-                    <i class="fa fa-eye"></i>
-                    View
-                  </button>
-                  <?php $showDownload = CRM_Utils_Array::value($rowID, $documentFileCount, 0) > 0 ? 'true' : 'false'; ?>
-                  <a class="btn btn-sm btn-default"
-                    <?php printf("ng-show='!document.loadingModalData && %s'", $showDownload); ?>
-                    target="_blank"
-                    ng-href="/civicrm/tasksassignments/file/zip?entityID=<?php print $rowID; ?>&entityTable=civicrm_activity"
-                  >
-                    <i class="fa fa-download"></i>
-                    Download
-                  </a>
-                <?php endif; ?>
-              </td>
-            </tr>
-          <?php endforeach; ?>
+        <?php foreach ($rows as $row_count => $row):
+          $mode = 'view';
+          $faIcon = 'fa-eye';
+          $label = 'View';
+          $rowID = strip_tags(CRM_Utils_Array::value('id', $row));
+          $status = CRM_Utils_Array::value('status_id', $row);
+
+          if($status === 'awaiting upload') {
+            $mode = 'edit';
+            $label = 'Upload';
+            $faIcon = 'fa-upload';
+          }
+
+          ?>
+          <!-- no results row -->
+          <tr id = "no-results-row" style="display:none;">
+            <td colspan="100%">There are no documents to display.</td>
+          </tr>
```

Comment:
I would have expected that too, but when you have a "no results behavior" in the view, and you have no results, the "no results text" is stored in $rows as 

```
$rows = [
  ["There are no documents to display."]
];
```
As you've probably noticed we don't need this text stored in the view anymore, because we're adding it in the template. However removing the "no results behaviour" from the view will result in an empty screen. 

I needed to move it to the template because of the mix of angular filters and drupal views going on here. Drupal views will show the no results text only when there are zero overall results. But for this ticket we need to show it when there are no results matching the selected filter too. I've fixed that an pushed my changes.

This comment did make me notice that I was putting the no results row in the wrong place. It should be outside the loop. Originally it was part of an if statement in the loop checking for `$row[0]` which would indicate no results, but since we always need it, it should be outside of the loop altogether.